pkg/controller/controller.go

```diff
@@ -155,8 +194,23 @@ const (
 	errorWithParameters         string = "ErrorWithParameters"
 )
 
+func (c *controller) reconcileBrokerKey(key string) error {
+	broker, err := c.brokerLister.Get(key)
+	if errors.IsNotFound(err) {
+		glog.Infof("Broker has been deleted %v", key)
+		return nil
+	}
+	if err != nil {
+		glog.Infof("Unable to retrieve broker %v from store: %v", key, err)
+		c.brokerQueue.Add(key)
+		return err
+	}
+
+	return c.reconcileBroker(broker)
+}
+
 // reconcileBroker is the control-loop that reconciles a Broker.
-func (c *controller) reconcileBroker(broker *v1alpha1.Broker) {
+func (c *controller) reconcileBroker(broker *v1alpha1.Broker) error {
```

Comment:
You would need to set up a message queue accessible to all controllers (network accessible) to which events were queued by a watching agent and dequeued by controllers.  I'm having flashbacks of setting up AMQP for Openstack...